library/std/src/sys/unix/ext/fs.rs

```diff
@@ -345,6 +345,30 @@ pub trait OpenOptionsExt {
     /// ```
     #[stable(feature = "open_options_ext", since = "1.10.0")]
     fn custom_flags(&mut self, flags: i32) -> &mut Self;
+
+    /// Get the flags as [`libc::c_int`].
+    ///
+    /// This method allows the reuse of the OpenOptions as flags argument for [`libc::open`].
+    ///
+    /// [`libc::c_int`]: https://docs.rs/libc/*/libc/type.c_int.html
+    /// [`libc::open`]: https://docs.rs/libc/*/libc/fn.open.html
+    ///
+    /// # Examples
+    ///
+    /// ```no_run
+    /// # #![feature(rustc_private)]
+    /// extern crate libc;
+    /// use std::ffi::CString;
+    /// use std::fs::OpenOptions;
+    /// use std::os::unix::fs::OpenOptionsExt;
+    ///
+    /// let mut options = OpenOptions::new();
+    /// options.write(true).read(true);
+    /// let file_name = CString::new("foo.txt").unwrap();
+    /// let file = unsafe { libc::open(file_name.as_c_str().as_ptr(), options.as_flags().unwrap()) };
+    /// ```
+    #[stable(feature = "open_options_ext_as_flags", since = "1.47.0")]
```

Comment:
I don't think this should be insta-stable. Usually new libs features go through at least a few cycles of nightly.
```suggestion
    #[unstable(feature = "open_options_ext_as_flags", issue = "none")]
```
Make sure to open a tracking issue and replace "none" with the issue number.